### PR TITLE
CORTX-31170: CSM user management and IAM users management API changes

### PR DESCRIPTION
**Pull request by abhijit1patil**
Signed-off-by: abhijit1patil <abhijit.1.patil@seagate.com>

# Problem Statement
- [CORTX-31170 JIRA ticket](https://jts.seagate.com/browse/CORTX-31170)

# Design
-  [Management API Changes](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1006305559/Management+API+Changes)

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Build and Deployment

- [Custom Build Job](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6117)
- Custom Build Images
   cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-6117-custom-ci 
   cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6117-custom-ci
   cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6117-custom-ci
- [Deployment Job](https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/4462/)
- 1 node Deployed VM

# Dev Manual Testing

- Manual Testing for CSM User endpoint
     - Endpoint: /api/v2/system/users
image
 
     - Endpoint:- /api/v2/system/users/{user_id}
image

- Manual Testing for Iam User endpoint
        - Endpoint:-/api/v2/iam/users
image

        - Endpoint:- /api/v2/iam/users/{uid}
image

        - Endpoint:- /api/v2/iam/caps/{uid}
image

        - Endpoint:- /api/v2/iam/quota/{uid}
image

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/core/controllers/users.py [hunk 1]
```diff
@@ -91,8 +91,8 @@ class CsmGetUsersSchema(Schema):
     role = fields.Str(default=None, missing=None)
 
 
-@CsmView._app_routes.view("/api/v1/csm/users")
-@CsmView._app_routes.view("/api/v2/csm/users")
+@CsmView._app_routes.view("/api/v1/users")
+@CsmView._app_routes.view("/api/v2/users")
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:95]: Change /api/v2/users -> /api/v2/system/users

Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:95]: Change /api/v2/users -> /api/v2/system/users

File: csm/core/controllers/users.py [hunk 2]
```diff
@@ -153,8 +153,8 @@ async def post(self):
         return CsmResponse(response, const.STATUS_CREATED)
 
 
-@CsmView._app_routes.view("/api/v1/csm/users/{user_id}")
-@CsmView._app_routes.view("/api/v2/csm/users/{user_id}")
+@CsmView._app_routes.view("/api/v1/users/{user_id}")
+@CsmView._app_routes.view("/api/v2/users/{user_id}")
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:157]: Change /api/v2/users/{user_id} -> /api/v2/system/users/{user_id}

Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:157]: Change /api/v2/users/{user_id} -> /api/v2/system/users/{user_id}

File: csm/core/controllers/users.py [hunk 3]
```diff
@@ -91,8 +91,8 @@ class CsmGetUsersSchema(Schema):
     role = fields.Str(default=None, missing=None)
 
 
-@CsmView._app_routes.view("/api/v1/csm/users")
-@CsmView._app_routes.view("/api/v2/csm/users")
+@CsmView._app_routes.view("/api/v1/users")
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:94]: Please do not change any v1 api's

Comment (pawankumarsrivastava) [inline csm/core/controllers/users.py:94]: Please do not change any v1 api's

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/controllers/users.py (hunk 1 above)
Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

File: csm/core/controllers/users.py (hunk 2 above)
Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

File: csm/core/controllers/users.py (hunk 3 above)
Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

Comment (abhijit1patil) [reply csm/core/controllers/users.py]: updated

**Review by rohitkolapkar — APPROVED** [review]

**Pull request merged**